<?xml version="1.0"?>
<launch>
    <!-- Select only one data type -->
    <arg name="using_gazebo_data" default="false" />
    <arg name="using_logging_data" default="false" />
    <arg name="using_device_data" default="true" />

    <!-- Aruco lib param -->
    <arg name="detector_params_path" default="$(find kuam_aruco_tracking)/config/detector_params.yml" />
    <arg name="calib_path" default="$(find kuam_aruco_tracking)/config/calib_sim.txt" unless="$(arg using_device_data)"/>
    <arg name="calib_path" default="$(find kuam_aruco_tracking)/config/calib.txt" if="$(arg using_device_data)"/>

    <arg name="usb_cam_logging_topic" default="/usb_cam/image_rect_color" if="$(arg using_logging_data)"/>
    <arg name="usb_cam_logging_topic" default="/usb_cam/image_rect_color" unless="$(arg using_logging_data)"/>

        <!-- Estimating method -->
        <!-- 0. Without filter -->
        <!-- 1. Moving average filter -->
        <!-- 2. Exponential moving average filter -->
    <arg name="estimating_method" default="2" />

    <include file="$(find kuam_aruco_tracking)/launch/aruco_tracking.launch">
        <arg name="calib_path" value="$(arg calib_path)" />
        <arg name="detector_params_path" value="$(arg detector_params_path)" />
        <arg name="estimating_method" value="$(arg estimating_method)" />
        <arg name="using_gazebo_data" value="$(arg using_gazebo_data)" />
        <arg name="using_logging_data" value="$(arg using_logging_data)" />
        <arg name="usb_cam_logging_topic" value="$(arg usb_cam_logging_topic)" />
    </include>

	<node pkg="kuam_aruco_tracking" type="aruco_visual_node" name="aruco_visual" />
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find kuam_aruco_tracking)/rviz/aruco_marker.rviz" required="true" />
</launch>
